weblate/addons/base.py

```diff
@@ -72,27 +74,38 @@ def get_identifier(cls):
         return cls.name
 
     @classmethod
-    def create_object(cls, component, **kwargs):
+    def create_object(
+        cls, *, component: Component = None, project: Project = None, **kwargs
```

Comment:
The type hints should indicate this is an optional arg:

```suggestion
        cls, *, component: Component | None = None, project: Project | None = None, **kwargs
```